src/ray/raylet/worker_pool.cc

```diff
@@ -918,92 +949,51 @@ void WorkerPool::TryKillingIdleWorkers() {
   RAY_CHECK(idle_of_all_languages_.size() == idle_of_all_languages_map_.size());
 }
 
+void WorkerPool::TriggerAsyncCallbacksForFailedWorkerStart(
```

Comment:
This is a bit subtle. What if we just sent the error to just 1 pending request? Or all?

Reply:
I don't think it would be correct to only send the error to 1 pending request, since you can have fewer starting workers than pending requests. If we did all, I'm a bit concerned we'll pingpong, e.g., fail all requests because there are too many workers starting, but then the workers exit before we can request them again.